BraveShared/Javascript/JSError.swift

```diff
@@ -0,0 +1,19 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import JavaScriptCore
+
+public struct JSError: Error, CustomStringConvertible {
```

Comment:
Since Error is a protocol you can use it with String Extension as: `extension String: Error {}` this line makes any String conform to Error. You can simply pass a String and need not create a struct with it.